namenode/src/main/java/com/ibm/crail/namenode/FileStore.java

```diff
@@ -40,11 +40,17 @@ public FileStore(Sequencer sequencer) throws IOException {
 	
 	public AbstractNode createNode(int fileComponent, CrailNodeType type, int storageClass, int locationClass) throws IOException {
 		if (type == CrailNodeType.DIRECTORY){
-			return new DirectoryBlocks(sequencer.getNextId(), fileComponent, CrailNodeType.DIRECTORY, storageClass, locationClass);
+			return new DirectoryBlocks(sequencer.getNextId(), fileComponent, type, storageClass, locationClass);
```

Comment:
Considering that there are more types to come - I don't think that this hand enumeration of types is scalable ;) So what we can do is 
1. make it possible to register new types in some sort of global registry 
2. make an interface to construct these objects since they all look alike in terms of parameters. 
3. That would simplify the logic where you just look up in table and xyz.construct(...), instead of a series of if-else.